config/initializers/extensions.rb

```diff
@@ -0,0 +1,8 @@
+class Array
+  # Use one_and_only instead of first when dealing with arrays that
+  # should only ever have one element. 
+  def one_and_only
+    raise "Not an array with a single element" unless size == 1
+    first
+  end
```

Comment:
For context, it is used here:

https://github.com/alphagov/collections/blob/use-content-store-for-browse-pages/app/models/second_level_browse_page.rb#L28

Elements in the `links` hash from the content-store are _always_ arrays, even if it makes no sense for them to be arrays and there can only ever be one of them.

If we write `links.active_top_level_browse_page.first` it reads very awkwardly because, there can't be a first or last of singular thing. It tells the developer that we can expect more items, and that they may be used somewhere else. There is also a risk that elements are added to the array. When that happens, the `active_top_level_browse_page` becomes meaningless and we should be warned about it.

`Array#one_and_only` tries to communicate clearly to the reader that only one element will be ever expected in the array. Ideally, this problem should be solved by the content-store by returning just a hash and not an array. But from discussions I've gathered that that is not an option. That's why I'm proposing this solution. 

Open to other viewpoints!

You're right about the zero-length array, it could be `unless size > 1`